redbot/cogs/mod/names.py

```diff
@@ -69,6 +69,88 @@ class ModInfo(MixinMeta):
             else:
                 await ctx.send(_("Done."))
 
+    def handle_custom(self, user):
+        a = [c for c in user.activities if c.type == discord.ActivityType.custom]
+        if not a:
+            return None, discord.ActivityType.custom
+        a = a[0]
+        c_status = None
+        if not a.name and not a.emoji:
+            return None, discord.ActivityType.custom
+        if not a.name:
+            c_status = self.bot.get_emoji(a.emoji.id)
+        if c_status:
+            pass
+        if a.name and a.emoji:
+            c_status = f"{a.emoji} {a.name}"
+        elif a.emoji and not c_status:
+            c_status = f"{a.emoji}"
+        elif a.name:
+            c_status = a.name
+        else:
+            c_status = None
+        return c_status, discord.ActivityType.custom
+
+    def handle_playing(self, user):
+        p_acts = [c for c in user.activities if c.type == discord.ActivityType.playing]
+        p_act = p_acts[0] if p_acts else None
+        act = p_act.name if p_act and p_act.name else None
+        return act, discord.ActivityType.playing
+
+    def handle_streaming(self, user):
+        s_acts = [c for c in user.activities if c.type == discord.ActivityType.streaming]
+        s_act = s_acts[0] if s_acts else None
+        act = (
+            f"[{s_act.name}{' | ' if s_act.game else ''}{s_act.game or ''}]({s_act.url})"
+            if s_act and s_act.name and hasattr(s_act, "game")
+            else s_act.name
+            if s_act and s_act.name
+            else None
+        )
+        return act, discord.ActivityType.streaming
+
+    def handle_listening(self, user):
+        l_acts = [c for c in user.activities if c.type == discord.ActivityType.listening]
+        l_act = l_acts[0] if l_acts else None
+        act = (
+            f"[{l_act.title}{' | ' if l_act.artists[0] else ''}{l_act.artists[0] or ''}](https://open.spotify.com/track/{l_act.track_id})"
```

Comment:
The text inside square brackets needs markdown escaping, you can use `discord.utils.escape_markdown()` function for that